### PR TITLE
Fix for issue "Delete dialog appears when pressing ctrl+alt+del on latest commit of a branch" with id 5636

### LINKED ISSUE #5636: Delete dialog appears when pressing ctrl+alt+del on latest commit of a branch
**Do you want to request a *feature* or report a *bug*?**
Bug.

**What is the current behavior?**
If the latest commit of a non-checked out branch is highlighted, and the main Git Extensions window has focus, then when you press ctrl+alt+del, the delete dialog appears.

Edit: The Windows ctrl+alt+del screen comes up and you have to close it before the delete dialog appears.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
1. Highlight the latest commit of a branch you do not have checked out.
2. Press ctrl+alt+del.

**What is the expected behavior?**
Delete key press should be ignored when pressing ctrl+alt+del.

**Environment you encounter the issue:**

- GitExtensions version: 2.51.04
- GIT version: N/A
- OS version: Windows 7 SP1
- .NET version: N/A

**Did this work in previous version of GitExtensions (which)?**
Not sure.

### PR DESCRIPTION
**Pull request by AbhiAgarwal192**
Fixes #5636

Changes proposed in this pull request:
- On clicking ctrl+alt+del , modifiers ctrl+alt were ignored and it was treated as if del key was fixed. With this fix if modifiers are present then del key is ignored.

 
Screenshots before and after (if PR changes UI):
- None

What did I do to test the code and ensure quality:
- Manual testing of the change.

Has been tested on (remove any that don't apply):
- GIT 2.11 and above: 2.18
- Windows 7 and above : windows 10.

### CODE REVIEW

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 1]
```diff
@@ -1255,7 +1255,11 @@ private void OnGridViewKeyUp(object sender, KeyEventArgs e)
 
                 case Keys.Delete:
                     {
-                        InitiateRefAction(
+                        // Check modifier detail if pressed.
+
+                        if (e.Modifiers == Keys.None)
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: Please invert `if` to reduce nesting.

Comment (AbhiAgarwal192) [reply GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: If i add the "if" clause before switch then i guess nesting is reduced.

**Comment by RussKie** [comment]
That will work too

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 2]
```diff
@@ -1241,6 +1241,13 @@ private void OnGridViewKeyUp(object sender, KeyEventArgs e)
             {
                 return;
             }
+			
+			// Check modifier detail if pressed.
+
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: unnecessary empty line

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 3]
```diff
@@ -1255,6 +1262,7 @@ private void OnGridViewKeyUp(object sender, KeyEventArgs e)
 
                 case Keys.Delete:
                     {
+
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: unnecessary empty line

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 4]
```diff
@@ -1273,6 +1281,7 @@ private void OnGridViewKeyUp(object sender, KeyEventArgs e)
                                 }
                             },
                             FormQuickGitRefSelector.Action.Delete);
+
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: unnecessary empty line

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 5]
```diff
@@ -1241,6 +1241,13 @@ private void OnGridViewKeyUp(object sender, KeyEventArgs e)
             {
                 return;
             }
+			
+			// Check modifier detail if pressed.
+			
+			if (e.Modifiers == Keys.None)
+            {
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: misindented

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 6]
```diff
@@ -1241,6 +1241,13 @@ private void OnGridViewKeyUp(object sender, KeyEventArgs e)
             {
                 return;
             }
+			
+			// Check modifier detail if pressed.
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: This comment does not carry any real value, please remove

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs (hunk 2 above)
Comment (AbhiAgarwal192) [reply GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: Fixed.

**Comment by RussKie** [comment]
Please remove the unnecessary empty lines and it will be good to go.

File: GitUI/Editor/FileViewer.cs [hunk 7]
```diff
@@ -216,7 +216,7 @@ public Encoding Encoding
                     }
                     else
                     {
-                        encodingToolStripComboBox.SelectedIndex = -1;
+                        encodingToolStripComboBox.SelectedIndex = 0;
```
Comment (gerhardol) [inline GitUI/Editor/FileViewer.cs]: Should be submitted in a separate PR, for #5575

Comment (gerhardol) [reply GitUI/Editor/FileViewer.cs]: I get exception at startup due to this:
System.ArgumentOutOfRangeException: InvalidArgument=Value på 0 är inte giltigt för SelectedIndex.
Parameternamn: SelectedIndex
   vid System.Windows.Forms.ComboBox.set_SelectedIndex(Int32 value)
   vid GitUI.Editor.FileViewer.<set_Encoding>b__64_0() i F:\dev\gc\gitextensions_4\GitUI\Editor\FileViewer.cs:rad 219
   vid GitUI.GitUIExtensions.<InvokeAsync>d__11.MoveNext() i F:\dev\gc\gitextensions_4\GitUI\GitUIExtensions.cs:rad 208
--- Slut på stackspårningen från föregående plats där ett undantag utlöstes ---
   vid System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   vid System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   vid System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   vid GitUI.ThreadHelper.<>c__DisplayClass13_0.<<FileAndForget>b__0>d.MoveNext() i F:\dev\gc\gitextensions_4\GitExtUtils\GitUI\ThreadHelper.cs:rad 101

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 8]
```diff
@@ -1242,6 +1242,13 @@ private void OnGridViewKeyUp(object sender, KeyEventArgs e)
                 return;
             }
 
+            // Check modifier detail if pressed.
+
+            if (e.Modifiers == Keys.None)
```
Comment (gerhardol) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: This check is incorrect, should be !=
DEL/F2 only works with modifier now

**Comment by AbhiAgarwal192** [comment]
Reverted the changes for #5575

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs (hunk 6 above)
Comment (AbhiAgarwal192) [reply GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: Done

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs (hunk 8 above)
Comment (AbhiAgarwal192) [reply GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: Done

File: GitUI/Editor/FileViewer.cs (hunk 7 above)
Comment (AbhiAgarwal192) [reply GitUI/Editor/FileViewer.cs]: Will fix this and submit another PR.

**Comment by RussKie** [comment]
The diff looks good, however there are too many commits. 
Please remove all irrelevant commits or squash them.

**Comment by gerhardol** [comment]
There is one comment for an unnecessary line too.
@AbhiAgarwal192 You may also agree to squash merge after fixing too.

**Comment by RussKie** [comment]
There are too many commits, I don't want to squash-merge it.

**Comment by GintasS** [comment]
This is an old PR and should be closed or set to a draft. @RussKie
